### PR TITLE
FW-6251 Add draftjs conversion script and HTML cleaning to relevant text fields

### PR DESCRIPTION
**Pull request by bronson-c**
### Description of Changes
- Adds a `convert_draftjs_to_html` command, which can be used to convert all draftjs content to html for a specific set of sites, or all sites
- Adds a `HTMLSanitizedTextField` which uses nh3 to clean html input on save and replaces all former draftjs fields with the HTMLSanitizedTextField
- Adds tests for html cleaning and draftjs conversion

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] ~~Admin Panel has been updated if models have been added or modified~~
- [x] Migrations have been updated and committed if applicable
- [x] ~~Insomnia workspace has been updated if applicable~~

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/management/commands/convert_draftjs_to_html.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            return False
+
+    def convert_draftjs_to_html(self, text):
+        if self.is_draftjs_content(text):
+            exporter = HTML()
+            return exporter.render(json.loads(text))
+        return text
+
+    def handle(self, *args, **options):
+        logger = logging.getLogger("convert_draftjs_to_html")
```
Comment (sarahfirstvoices) [inline firstvoices/backend/management/commands/convert_draftjs_to_html.py:41]: To make our logs more predictable, please use `logging.getLogger(__name__)` and just leave it at the default level

File: firstvoices/backend/management/commands/convert_draftjs_to_html.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        logger.setLevel(logging.INFO)
+
+        if options.get("site_slugs"):
+            site_slug_list = [
+                s.strip() for s in options.get("site_slugs", "").split(",")
+            ]
+            sites = Site.objects.filter(slug__in=site_slug_list)
+            if not sites:
+                logger.info("No sites with the provided slug(s) found.")
```
Comment (sarahfirstvoices) [inline firstvoices/backend/management/commands/convert_draftjs_to_html.py:50]: I think this one could be a warning

File: firstvoices/backend/management/commands/convert_draftjs_to_html.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                return
+        else:
+            sites = Site.objects.all()
+
+        for site in sites:
+            logger.info(f"Converting draftjs content to html for site {site.slug}...")
```
Comment (sarahfirstvoices) [inline firstvoices/backend/management/commands/convert_draftjs_to_html.py:56]: Looking at this and the rest of the logs in this method-- I think they should all be debug level logs

File: firstvoices/backend/models/base.py [hunk 4]
```diff
@@ -124,6 +125,19 @@ def get_prep_value(self, value):
         return value
 
 
+class HTMLSanitizedTextField(models.TextField):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/models/base.py:128]: Proposed rename: SanitizedHtmlField

File: firstvoices/backend/tests/test_models/test_song_model.py [hunk 5]
```diff
@@ -9,3 +11,15 @@ class TestSongModel(RelatedVideoLinksValidationMixin):
 
     def create_instance_with_related_video_links(self, site, related_video_links):
         return SongFactory.create(site=site, related_video_links=related_video_links)
+
+    @pytest.mark.django_db
+    def test_html_cleaning_fields(self):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_models/test_song_model.py:16]: These tests don't need to go on the models-- all the logic is in the field, so please just add one test file with tests for the field.

Comment (sarahfirstvoices) [reply firstvoices/backend/tests/test_models/test_song_model.py:16]: When you do that, can you please add a test case where we save a non-html string that has some html-related characters in it? I am thinking about the widget settings using the HTML fi1eld for every value. So e.g., the text could be "Why is 1 < 2 ?" or "The French person said << quoi ?? >>"

**Review by sarahfirstvoices — CHANGES_REQUESTED** [review]

File: firstvoices/backend/management/commands/convert_draftjs_to_html.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+            sites = Site.objects.all()
+
+        for site in sites:
+            logger.debug(f"Converting draftjs content to html for site {site.slug}...")
+            songs_to_convert = Song.objects.filter(site=site)
+            stories_to_convert = Story.objects.filter(site=site)
+            story_pages_to_convert = StoryPage.objects.filter(site=site)
+            site_widgets = SiteWidget.objects.filter(site=site)
+            widget_settings_to_convert = WidgetSettings.objects.filter(
+                widget__in=site_widgets, key="textWithFormtting"
```
Comment (gmcauliffe) [inline firstvoices/backend/management/commands/convert_draftjs_to_html.py:61]: Spotted a typo: "textWithForm**a**tting"

Comment (bronson-c) [reply firstvoices/backend/management/commands/convert_draftjs_to_html.py]: fixed! Nice catch

**Review by sarahfirstvoices — APPROVED** [review]

**Pull request merged**